src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js

```diff
@@ -38,8 +39,9 @@ class ScholarshipListDataHandler extends ListDataHandler {
    * The total number of scholarship stored in backend.
    */
   async getTotalNumber() {
-    // TODO: fetch from backend.
-    return 35;  // Presetting page length not supported yet.
+    const data = await fetch(DATA_SIZE_ENDPOINT);
+    const number = await data.text();
+    return parseInt(number);
```

Comment:
add base to parseInt. Also, how is NaN handled?

`return parseInt(number, 10);`

Reply:
good catch! thanks!